src/handlers/flow/FlowStreamHandler.h

```diff
@@ -0,0 +1,222 @@
+    {
+        set_event_rate_info("flows", {"rates", "pps_total"}, "Rate of all packets (combined ingress and egress) in packets per second");
+        set_num_events_info("flows", {"total"}, "Total packets processed");
+        set_num_sample_info("flows", {"deep_samples"}, "Total packets that were sampled for deep inspection");
+    }
+
+    // get a copy of the counters
+    counters counters() const
+    {
+        std::shared_lock lock(_mutex);
+        return _counters;
+    }
+
+    // visor::AbstractMetricsBucket
+    void specialized_merge(const AbstractMetricsBucket &other) override;
+    void to_json(json &j) const override;
+    void to_prometheus(std::stringstream &out, Metric::LabelMap add_labels = {}) const override;
+
+    // must be thread safe as it is called from time window maintenance thread
+    void on_set_read_only() override
+    {
+        // stop rate collection
+        _rate.cancel();
+        _throughput.cancel();
+    }
+
+    void process_flow(bool deep, const FlowPacket &payload);
+};
+
+class FlowMetricsManager final : public visor::AbstractMetricsManager<FlowMetricsBucket>
+{
+public:
+    FlowMetricsManager(const Configurable *window_config)
+        : visor::AbstractMetricsManager<FlowMetricsBucket>(window_config)
+    {
+    }
+
+    void process_flow(const FlowPacket &payload);
+};
+
+class FlowStreamHandler final : public visor::StreamMetricsHandler<FlowMetricsManager>
+{
+
+    // the input stream sources we support (only one will be in use at a time)
+    MockInputStream *_mock_stream{nullptr};
+    FlowInputStream *_flow_stream{nullptr};
+
+    sigslot::connection _sflow_connection;
+    sigslot::connection _netflow_connection;
+
+    std::vector<Ipv4Subnet> _IPv4_host_list;
+    std::vector<Ipv6Subnet> _IPv6_host_list;
+
+    enum Filters {
+        OnlyHosts,
+        FiltersMAX
+    };
+    std::bitset<Filters::FiltersMAX> _f_enabled;
+
+    static const inline StreamMetricsHandler::GroupDefType _group_defs = {
+        {"cardinality", group::FlowMetrics::Cardinality},
+        {"counters", group::FlowMetrics::Counters},
+        {"top_geo", group::FlowMetrics::TopGeo},
+        {"top_by_bytes", group::FlowMetrics::TopByBytes},
+        {"top_by_packets", group::FlowMetrics::TopByPackets}};
+
+    void process_sflow_cb(const SFSample &);
+    void process_netflow_cb(const NFSample &);
+    void set_start_tstamp(timespec stamp);
+    void set_end_tstamp(timespec stamp);
+
+    void _parse_host_specs(const std::vector<std::string> &host_list);
+    bool _match_subnet(uint32_t ipv4 = 0, const uint8_t* ipv6 = nullptr);
+    bool _filtering(const FlowData &flow);
+
+public:
+    FlowStreamHandler(const std::string &name, InputStream *stream, const Configurable *window_config, StreamHandler *handler = nullptr);
+    ~FlowStreamHandler() override;
+
+    // visor::AbstractModule
+    std::string schema_key() const override
+    {
+        return "flows";
```

Comment:
can we make this a static const somewhere and reference it, rather than a string?

Reply:
where should we do that? all the handlers are using in this way. should be inside the handler itself due to dynamic loading plugins

Reply:
yes for internal use in the handler, for example in all the metrics definitions